skglm/utils.py

```diff
@@ -1,10 +1,93 @@
+from functools import lru_cache
```

Comment:
Nitpick, but I would tend to put these functions into a separated `utils_jit.py` or `utils_numba.py`

Reply:
this file indeed deserves a refactoring, but I'd do it later: there are math utils, the Anderson class, etc that should be put in separate files